### PR TITLE
[DO NOT MERGE]stub file for API review

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/out/_azure_appconfiguration_client.pyi [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    def list_configuration_settings(self, keys: list=..., labels: list=..., **kwargs: dict) -> azure.core.paging.ItemPaged[ConfigurationSetting]: ...
+    def get_configuration_setting(self, key: str, label: str=..., etag: str=..., match_condition: MatchConditions=..., **kwargs: dict) -> ConfigurationSetting: ...
+    def add_configuration_setting(self, configuration_setting: ConfigurationSetting, **kwargs: dict) -> ConfigurationSetting: ...
+    def set_configuration_setting(self, configuration_setting: ConfigurationSetting, match_condition: MatchConditions=..., **kwargs: dict) -> ConfigurationSetting: ...
+    def delete_configuration_setting(self, key: str, label: str=..., etag: str=..., match_condition: MatchConditions=..., **kwargs: dict) -> ConfigurationSetting: ...
+    def list_revisions(self, keys: list=..., labels: list=..., **kwargs: dict) -> azure.core.paging.ItemPaged[ConfigurationSetting]: ...
+    def set_read_only(self, configuration_setting: ConfigurationSetting, **kwargs: dict) -> ConfigurationSetting: ...
+    def clear_read_only(self, configuration_setting: ConfigurationSetting, **kwargs: dict) -> ConfigurationSetting: ...
```
Comment (bryevdv) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/out/_azure_appconfiguration_client.pyi]: Some of these are `**kwargs: Any` and others are `**kwargs: dict` is this intentional?

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/out/_azure_appconfiguration_client.pyi [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+from ._azure_appconfiguration_error import ResourceReadOnlyError
+from ._azure_appconfiguration_requests import AppConfigRequestsCredentialsPolicy
+from ._generated import AzureAppConfiguration
+from ._generated._configuration import AzureAppConfigurationConfiguration
+from ._generated.models import ErrorException, KeyValue
+from ._models import ConfigurationSetting
+from ._user_agent import USER_AGENT
+from ._utils import escape_and_tostr, get_endpoint_from_connection_string, prep_if_match, prep_if_none_match
+from azure.core import MatchConditions
+from typing import Any
```
Comment (bryevdv) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/out/_azure_appconfiguration_client.pyi]: Should be behind `if TYPE_CHECKING:` ? (same below)

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/out/aio/_azure_configuration_client_async.pyi [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+from .._generated.aio._configuration_async import AzureAppConfigurationConfiguration
+from .._generated.models import ErrorException, KeyValue
+from .._models import ConfigurationSetting
+from .._user_agent import USER_AGENT
+from .._utils import escape_and_tostr, get_endpoint_from_connection_string, prep_if_match, prep_if_none_match
+from azure.core import MatchConditions
+from typing import Any
+
+class AzureAppConfigurationClient:
+    config: Any = ...
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/out/aio/_azure_configuration_client_async.pyi]: Config must be private and not exposed to customer.

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/out/aio/_azure_configuration_client_async.pyi [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    def __init__(self, base_url: str, credential: AppConfigConnectionStringCredential, **kwargs: Any) -> None: ...
+    @classmethod
+    def from_connection_string(cls: Any, connection_string: str, **kwargs: Any) -> AzureAppConfigurationClient: ...
+    def list_configuration_settings(self, keys: list=..., labels: list=..., **kwargs: dict) -> azure.core.paging.ItemPaged[ConfigurationSetting]: ...
+    async def get_configuration_setting(self, key: str, label: str=..., etag: str=..., match_condition: MatchConditions=..., **kwargs: dict) -> ConfigurationSetting: ...
+    async def add_configuration_setting(self, configuration_setting: ConfigurationSetting, **kwargs: dict) -> ConfigurationSetting: ...
+    async def set_configuration_setting(self, configuration_setting: ConfigurationSetting, match_condition: MatchConditions=..., **kwargs: dict) -> ConfigurationSetting: ...
+    async def delete_configuration_setting(self, key: str, label: str=..., etag: str=..., match_condition: MatchConditions=..., **kwargs: dict) -> ConfigurationSetting: ...
+    def list_revisions(self, keys: list=..., labels: list=..., **kwargs: dict) -> azure.core.paging.ItemPaged[ConfigurationSetting]: ...
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/out/aio/_azure_configuration_client_async.pyi]: Should be an AsyncItemPaged

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/out/_azure_appconfiguration_client.pyi (hunk 1 above)
Comment (xiangyan99) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/out/_azure_appconfiguration_client.pyi]: Good catch!

File: sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/out/_azure_appconfiguration_client.pyi [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    @classmethod
+    def from_connection_string(cls: Any, connection_string: string, **kwargs: dict) -> AzureAppConfigurationClient: ...
+    def list_configuration_settings(self, keys: list=..., labels: list=..., **kwargs: dict) -> azure.core.paging.ItemPaged[ConfigurationSetting]: ...
+    def get_configuration_setting(self, key: str, label: str=..., etag: str=..., match_condition: MatchConditions=..., **kwargs: dict) -> ConfigurationSetting: ...
+    def add_configuration_setting(self, configuration_setting: ConfigurationSetting, **kwargs: dict) -> ConfigurationSetting: ...
+    def set_configuration_setting(self, configuration_setting: ConfigurationSetting, match_condition: MatchConditions=..., **kwargs: dict) -> ConfigurationSetting: ...
+    def delete_configuration_setting(self, key: str, label: str=..., etag: str=..., match_condition: MatchConditions=..., **kwargs: dict) -> ConfigurationSetting: ...
+    def list_revisions(self, keys: list=..., labels: list=..., **kwargs: dict) -> azure.core.paging.ItemPaged[ConfigurationSetting]: ...
+    def set_read_only(self, configuration_setting: ConfigurationSetting, **kwargs: dict) -> ConfigurationSetting: ...
```
Comment (johanste) [inline sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/out/_azure_appconfiguration_client.pyi]: Does this take name/label in addition to configuration_setting?

Comment (xiangyan99) [reply sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/out/_azure_appconfiguration_client.pyi]: My thought is if in the future, we want to support conditional set_read_only, I would like the API more similar to set_configuration_setting
```python
def set_configuration_setting(self, configuration_setting: ConfigurationSetting, match_condition: MatchConditions=..., **kwargs: dict)
```
than delete_configuration_setting
```python
def delete_configuration_setting(self, key: str, label: str=..., etag: str=..., match_condition: MatchConditions=..., **kwargs: dict)
```

Neither will be a breaking change and I am open to either one.

**Comment by lmazuel** [comment]
@xiangyan99 closing since I think it's done, feel free to re-open if necessary

**Pull request closed without merging**